library/src/test/java/com/getstream/sdk/chat/rest/ClientTest.java

```diff
@@ -853,6 +857,80 @@ void queryChannelsSuccessTest() {
+        ChannelState response = new ChannelState();
+        response.setMessages(messages);
+        response.setChannel(channelResponse);
+        QueryChannelCallback callback = mock(QueryChannelCallback.class);
+
+        Channel channel = new Channel(client, TEST_CHANNEL_TYPE, TEST_CHANNEL_ID, extra);
+        channel.setChannelState(new ChannelState(channel));
+
+        when(apiService.queryChannel(any(), any(), any(), any(), any(), any()))
+                .thenReturn(CallFake.buildSuccess(response));
+        ChannelQueryRequest request = new ChannelQueryRequest()
+                .withPresence()
+                .withWatch();
+
+        client.queryChannel(channel, request, callback);
+
+        verify(apiService).queryChannel(eq(TEST_CHANNEL_TYPE), eq(TEST_CHANNEL_ID), eq(TEST_API_KEY),
+                eq(TEST_USER_ID), eq(TEST_CLIENT_ID), argThat(argument ->
+                        argument.isWatch()
+                                && argument.isPresence()
+                                && argument.getData().get("name").equals(testChannelName)));
+        verify(callback).onSuccess(response);
+        verify(storage).insertMessagesForChannel(channel, response.getMessages());
+    }
+
+    @Test
+    void queryChannelWithoutIdSuccessTest() {
+        ArrayList<Message> messages = new ArrayList<>();
+        messages.add(new Message());
+        HashMap<String, Object> extra = new HashMap<>();
+        String testChannelName = "testName";
+        extra.put("name", testChannelName);
+
+        Channel channelResponse = new Channel(client, TEST_CHANNEL_TYPE, TEST_CHANNEL_ID, extra);
+        ChannelState response = new ChannelState();
+        response.setMessages(messages);
+        response.setChannel(channelResponse);
+        QueryChannelCallback callback = mock(QueryChannelCallback.class);
+
+        Channel channel = new Channel(client, TEST_CHANNEL_TYPE, null, extra);
+        channel.setChannelState(new ChannelState(channel));
+
+        when(apiService.queryChannel(any(), any(), any(), any(), any()))
+                .thenReturn(CallFake.buildSuccess(response));
+        ChannelQueryRequest request = new ChannelQueryRequest()
+                .withPresence()
+                .withWatch();
+
+        client.queryChannel(channel, request, callback);
+
+        verify(apiService).queryChannel(eq(TEST_CHANNEL_TYPE), eq(TEST_API_KEY),
+                eq(TEST_USER_ID), eq(TEST_CLIENT_ID), argThat(argument ->
+                        argument.isWatch()
+                                && argument.isPresence()
+                                && argument.getData().get("name").equals(testChannelName)));
+        verify(callback).onSuccess(response);
+        verify(storage).insertMessagesForChannel(channel, response.getMessages());
+    }
+
+    @Test
+    void disconnectTest() {
+        client.disconnect();
+        verify(webSocketService).disconnect();
```

Comment:
It's also important to test that state is cleared:
```java
tokenProvider = null;
fetchingToken = false;
cacheUserToken = null;
... and all other values
```
That's quite crucial method, so we need to make sure that everything is clear when it's called.

Reply:
All of these fields are private. We can't test private fields in a simple way(only reflection or write getters only for testing but this way we break encapsulation principles). And as far as I understand we shouldn't test private methods/fields. Private variables are part of the implementation, tests should focus on behavior rather than implementation details.
Please correct me if I wrong.

Reply:
The easiest way to test the method is to make these fields package private and move `ClientTest` to `com.getstream.sdk.chat.rest.core`, currently it's `com.getstream.sdk.chat.rest`. Then tests will see package private fields.

About testing private fields. I agree that usually private fields shouldn't be touched. But `Client` is crucial part of SDK, which has complicated state and the higher coverage of `Client` we have, the lower chance of fatal issue like accidental line removal or some code refactoring.

So just by moving `ClientTest` to proper package and making package private at least this fields:
```
tokenProvider
fetchingToken
cacheUserToken
activeChannelMap
```
We're not breaking encapsulation, not using reflection and getting extra safety.

Reply:
updated according to your suggestions